src/pngimage.cpp

```diff
@@ -469,6 +469,10 @@ namespace Exiv2 {
                     profileName_ = std::string(reinterpret_cast<char *>(chunkData.pData_), iccOffset-1);
                     ++iccOffset; // +1 = 'compressed' flag
 
+                    if (iccOffset > chunkLength) {
```

Comment:
FYI (in case you choose continue to contribute): we've got `enforce` (stolen from D) for these `if (cond) { throw exception; }` cases.